providers/class-two-factor-totp.php

```diff
@@ -56,6 +56,35 @@ protected function __construct() {
 		return parent::__construct();
 	}
 
+	/**
+	 * Timestamp returned by time()
+	 *
+	 * @var int $now
+	 */
+	private static $now;
+
+	/**
+	 * Override time() in the current object for testing.
+	 *
+	 * @return int
+	 */
+	private static function time() {
+		if ( empty( self::$now ) || null === self::$now ) {
+			return time();
+		}
+
+		return self::$now;
+	}
+
+	/**
+	 * Set up the internal state of time() invocations for deterministic generation.
+	 *
+	 * @param int $now Timestamp to use when overriding time().
+	 */
+	public static function __set_time( $now ) {
```

Comment:
It this necessary strictly for unit tests?

Reply:
Correct. The tests need to explicitly pass in the time in order to deterministically generate a TOTP.